providers/com/peerj/normalizer.py

```diff
@@ -0,0 +1,125 @@
+
+    def get_link_type(self, link):
+        if 'dx.doi.org' in link:
+            return 'doi'
+        if 'peerj.com' in link:
+            return 'provider'
+        return 'misc'
+
+class ThroughLinks(Parser):
+    link = Delegate(Link, ctx)
+
+class Tag(Parser):
+    name = ctx
+
+class ThroughTags(Parser):
+    tag = (Delegate(Tag), ctx)
+
+class Publisher(Parser):
+    name = ctx
+
+class Institution(Parser):
+    name = RunPython('get_author_institute', ctx)
+
+    def get_author_institute(self, context):
+        # read into a set while preserving order and passed back to erase duplicates
+        seen = set();
+        if 'author_institution' in context:
+            return [x for x in context['author_institution'] if x not in seen and not seen.add(x)]
+        else:
+            return [x for x in context['author_institutions'].split('; ') if x not in seen and not seen.add(x)]
+
+class Institutions(Parser):
+    name = ctx
+
+class Association(Parser):
+    pass # entity = Delegate(Publisher, ctx)
+
+class Person(Parser):
+    given_name = ParseName(ctx).first
+    family_name = ParseName(ctx).last
+
+class Contributor(Parser):
+    person = Delegate(Person, ctx)
+    cited_name = ctx
+    order_cited = ctx('index')
+
+class CreativeWork(Parser):
+    title = ctx.title
+    description = Maybe(ctx, 'description-html')
+    contributors = Map(Delegate(Contributor), ctx.author)
+    links = Concat(
+        Map(Delegate(ThroughLinks), ctx.pdf_url),
+        Map(Delegate(ThroughLinks), RunPython('format_doi', ctx.doi)),
+        Delegate(ThroughLinks.using(link=Delegate(ISSN)), Maybe(ctx, 'issn')),
+        Map(Delegate(ThroughLinks), ctx.fulltext_html_url)
+    )
+    publishers = Map(
+        Delegate(Association.using(entity=Delegate(Publisher))),
+        ctx.publisher
+    )
+    institutions = Map(
+            Delegate(Association.using(entity=Delegate(Institution))),
+            ctx
+        )
+    subject = Delegate(Tag, ctx.subjects[0])
+    date_created = RunPython('parse_date', ctx.date)
+    date_published = RunPython('parse_date', ctx.date)
+    language = ctx.language
+    tags = Concat(
+        Delegate(Tag, ctx.subjects),
+        Map(Delegate(ThroughTags), Maybe(ctx, 'keywords'))
+    )
+    class Extra:
+        modified = RunPython('parse_date', ctx.date)
+        subjects = Maybe(ctx, 'subjects')
+        affiliations = Map(
+            Delegate(Association.using(entity=Delegate(Institution))),
+            ctx
+        )
+        identifiers = ctx.identifiers
+        volume = ctx.volume
+        emails = Maybe(ctx, 'author_email')
+        journal_title = Try('journal_title', ctx)
+        journal_abbrev = Try('journal_abbrev', ctx)
+        description_nohtml = ctx.description
+
+    def format_doi(self, doi):
+        return format_doi_as_url(self, doi)
+
+    def parse_date(self, date_str):
+        return arrow.get(dateparser.parse(date_str)).to('UTC').isoformat()
+
+class PeerJNormalizer(Normalizer):
```

Comment:
This class is superfluous in the current state. You only have to define it if you're going to be swapping out the root parse or deserializing the data in a specific way.